src/s2wsjson/common/number_range.py

```diff
@@ -16,3 +16,12 @@ def validate_start_end_order(cls, values: dict[str, Any]) -> dict[str, Any]:
             raise ValueError(cls, 'start_of_range should not be higher than end_of_range')
 
         return values
+
+    def __hash__(self):
+        return hash(f'{self.start_of_range}|{self.end_of_range}')
+
+    def __eq__(self, other):
+        if isinstance(other, NumberRange):
+            return self.start_of_range == other.start_of_range and self.end_of_range and other.end_of_range
```

Comment:
A shorter form would be `return other.__hash__() == self.__hash__()`

Reply:
While I agree it is definitely shorter and does not require any changes in the future if new fields are added (however unlikely), there is a performance consideration. The hash implementation requires generating a new string. If each `==` first transforms any NumberRange to a `str`,  I expect it to be a lot slower than the current condition which checks both floats on equality.

I am unsure if we should treat performance considerations heavily at the moment. What do you think?

Reply:
Btw, I just noticed a bug. The `end_of_range` is not checked for equality. Fixing now xD

Reply:
Fixed!